### PR TITLE
Add configurable CorrDiff regression loss selection

### PR DESCRIPTION
**Pull request by younes-abid**
# PhysicsNeMo Pull Request

## Description
This PR adds configurable regression loss selection for the CorrDiff training
path when using `regression` and `lt_aware_regression` model names.

### What changed
- Added a configurable `loss` field in CorrDiff regression model configs.
- Added loss selector wiring in `examples/weather/corrdiff/train.py`, with
   `RegressionLoss` as default to preserve existing behavior.
- Added robust selectable regression loss variants:
   - `RegressionHuberLoss`
   - `RegressionCharbonnierLoss`
   - `RegressionEdgeAwareLoss`
   - `RegressionShiftTolerantLoss`
   - `RegressionHybridStructuralLoss`
- Exported the new loss classes through the diffusion metrics public API.
- Added tests covering the new selectable regression losses.
- Added a changelog entry for this user-visible feature.

### Why
This improves flexibility for CorrDiff regression training by allowing loss
function selection from config without changing code, while preserving backward
compatibility through the existing default loss.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request. 
- [x] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Dependencies

No new runtime dependencies were added.

## Review Process

I understand this PR will be reviewed by the PhysicsNeMo team and that
AI-generated comments are advisory.

### CODE REVIEW

File: physicsnemo/diffusion/metrics/legacy_losses.py [hunk 1]
```diff
@@ -553,6 +557,318 @@ def __call__(
...
+        )
+
+        all_losses = []
+        for dx, dy in self._candidate_shifts():
+            y_shifted, valid_mask = _shift_tensor_with_mask(y, dx=dx, dy=dy)
+            candidate = _charbonnier(pred - y_shifted, self.eps)
+            candidate = candidate * valid_mask + (1.0 - valid_mask) * 1e6
+            all_losses.append(candidate)
+
+        return torch.min(torch.stack(all_losses, dim=0), dim=0).values
```
[the remark this answers is not in the document]
Comment (younes-abid) [reply physicsnemo/diffusion/metrics/legacy_losses.py]: Nice catch I will remove this loss
